rmf_task/src/rmf_task/BackupFileManager.cpp

```diff
@@ -130,16 +147,25 @@ class BackupFileManager::Robot::Implementation
 private:
   void write(const std::string& state)
   {
-    throw std::runtime_error(
-      "[BackupFileManager::Robot::Implementation::write] not implemented yet");
+    std::ofstream pre_backup;
+    const std::string pre_backup_file_path = robot_directory.string() + "/" + pre_backup_file_name;
```

Comment:
It's shorter and more idiomatic to write this as:
```
const std::string pre_backup_file_path = robot_directory / pre_backup_file_name;
```

You can do the same for line 152 as well.